gnmi_server/server.go

```diff
@@ -268,6 +269,13 @@ func authenticate(config *Config, ctx context.Context) (context.Context, error)
 		if err == nil {
 			success = true
 		}
+		// role must be readwrite to support write access
+		if writeAccess && len(rc.Auth.Roles) != 0 {
```

Comment:
Check Roles length is not necessary, because it's already checked here:
https://github.com/sonic-net/sonic-gnmi/blob/c6f36b9bbe0360714df330f15fbdad0b88cec5a0/gnmi_server/clientCertAuth.go#L243

If you want only check roles for RW operation, you may also need check above code.

Suggest change to:

if writeAccess {
...
}